diff -r 2a43e83684d5 src/foreign/zthread/include/zthread/Guard.h
--- a/src/foreign/zthread/include/zthread/Guard.h	Sat Aug 10 11:31:46 2013 +0200
+++ b/src/foreign/zthread/include/zthread/Guard.h	Sun Aug 18 09:46:43 2013 +0200
@@ -108,7 +108,7 @@
   }
 
   template <class LockType>
-  static void createScope(LockHolder<LockType>& l, unsigned long ms) {
+  static bool createScope(LockHolder<LockType>& l, unsigned long ms) {
 
     if(Scope1::createScope(l, ms))
       if(!Scope2::createScope(l, ms)) {
--- a/src/foreign/vigra/vigra/cachedfileimage.hxx.orig	2012-08-18 09:22:30.000000000 -0600
+++ b/src/foreign/vigra/vigra/cachedfileimage.hxx	2012-08-18 09:33:40.000000000 -0600
@@ -555,7 +555,7 @@
 class NotifyingDirectionSelector<UnstridedArrayTag, T, Notify>
 {
 #ifdef __GNUC__
-friend class Notify::self_type;
+friend class Notify::CachedFileImageIteratorBase;
 #else
 friend typename Notify::self_type;
 #endif
